src/main/kotlin/org/wa/rceditor/application/view/fragments/StringCell.kt

```diff
@@ -1,36 +1,33 @@
 package org.wa.rceditor.application.view.fragments
 
+import javafx.beans.property.StringProperty
 import javafx.scene.layout.Priority
 import org.wa.rceditor.application.Styles
-import org.wa.rceditor.application.model.RelationItem
-import org.wa.rceditor.application.model.RelationItemModel
-import org.wa.rceditor.application.viewmodel.MainViewModel
 import tornadofx.*
 
-class RelationItemFragment: ListCellFragment<RelationItem>() {
-    private val viewModel by inject<MainViewModel>()
-    private val relation = RelationItemModel(itemProperty)
+class StringCell: ListCellFragment<StringProperty>() {
+    private val contributor = ItemViewModel(itemProperty = itemProperty)
+            .bind(autocommit = true) { item }
```

Comment:
**TL;DR**
Not sure you need this ItemViewModel especially since you have autocommit set to true.

**More explanation**
In the TFX design, a `ViewModel` is designed to be a buffer between the view and the data (usually also consisting of FX properties). In the view model, TFX expects you to bind view model properties to the data model properties. However, if you set autocommit to false (default, I think) changes that the view makes to the view model properties will not be committed (saved) to the data model properties. Because the changes are not committed immediately, the view model lets you do rollbacks since it hasn't actually changed the data model until you commit.

Like you did here, you can also tell the view model (`ItemViewModel` in this case) to autocommit the properties, meaning the changes are propagated back to the data model (the `item` in this case) automatically without an explicit commit action. Because of that, any changes to `contributor` will automatically change `item`. This is fine, but it means that adding an ItemViewModel here is probably redundant since you aren't leveraging the power of the view model. You can switch to just using `item` throughout this class instead of using `contributor`.

Now, to understand why we might actually want to use an ItemViewModel here, we could decide that we only want to commit the updated string property value when the user finishes editing the list cell. Without a view model (or with autocommit = true, as it is now), if I switch to editing mode, clear the text field, and start typing "M", "a", then "t", the value in the property will change for each letter, updating the values all the way back to the property in the view model. So the viewModel property might see this sequence of changes `"Max" -> "" -> "M" -> "Ma" -> "Mat" -> "Matt"` as I type my name. If that is something we don't want, we can fix it by using an ItemViewModel like you have, but turn off autocommit. Then in the onSave (or onAccept, I can't remember the name of the function to override in the ListCellFragment) you can call something like `itemViewModel.commit()`. Then the new value is assigned to the `item` field. So the viewModel would see something like: `"Max" -> (user types new name and leaves the editing state) -> "Matthew"`. The advantage of this is that if the user cancels instead of accepting the edit, the view model sees no change and still has the old value (which is correct since the edits were canceled, and the `itemViewModel.commit()` was not called).

All that to say, in this use case, you can probably just use `item` instead of and ItemViewModel and `contributor`.